### PR TITLE
[ONNX] Modified var_mean symbolic to support more combinations of dims

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Based on existing implementation of var_mean, values of dim have to be sequential and start with zero. The formats listed below are cause scenarios with incompatible dimension for the Sub node.
-> dim[1, 2]
-> dim[0, 2]
-> dim[2, 0]

The changes in this PR allow such formats to be supported in var_mean

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -2311,11 +2311,12 @@ def _var_mean(g, input, dim, unbiased, keepdim):
         num_elements = numel(g, input)
     else:
         mean = g.op("ReduceMean", input, axes_i=dim, keepdims_i=keepdim)
+        t_mean = g.op("ReduceMean", input, axes_i=dim, keepdims_i=1)
         redudced_dims = g.op("Shape", input)
         # dim could contain one or multiple dimensions
         redudced_dims = g.op("Gather", redudced_dims, g.op("Constant", value_t=torch.tensor(dim)), axis_i=0)
         num_elements = g.op("ReduceProd", redudced_dims, keepdims_i=0)
-    sub_v = g.op("Sub", input, mean)
+    sub_v = g.op("Sub", input, t_mean)
```
Comment (jiafatom) [inline torch/onnx/symbolic_opset9.py:2319]: (1) Why `t_mean` is only in `else` branch? it need also be in `if` branch, right? 
(2) if `mean` is replaced, we need delete it from `else` branch.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2311,11 +2311,12 @@ def _var_mean(g, input, dim, unbiased, keepdim):
         num_elements = numel(g, input)
     else:
         mean = g.op("ReduceMean", input, axes_i=dim, keepdims_i=keepdim)
+        t_mean = g.op("ReduceMean", input, axes_i=dim, keepdims_i=1)
```
Comment (jiafatom) [inline torch/onnx/symbolic_opset9.py:2314]: Don't quite get the logic here why this change can resolve this problem. The only change here is for `keepdims`, seems not change the logic very much...

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2319]: Thanks, added to if branch

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2314]: For example ->
if the input shape is (x,y,z) and dim values are (1,2) and keepdims is set to 0, the resulting shape of the reduceMean node is (x). Now when this node is passed to the Sub op, the multi-directional broadcasting converts this shape to (1,1,x). 
So essentially we are subtracting shapes (x,y,z) and (1,1,x). Hence we are setting keepdims to 1 always to avoid this mismatch in dimensions for the sub op.
But for cases in which user has passed keepdims=0, the mean value would be of wrong dimensions, hence the difference in t_mean and mean.

**Review by jiafatom — APPROVED** [review]
LGTM!

**Comment by SplitInfinity** [comment]
@shubhambhokare1 Are the code coverage warnings out of date? Have you already addressed them?

**Comment by SplitInfinity** [comment]
Ping @shubhambhokare1

**Comment by SplitInfinity** [comment]
Ping @shubhambhokare1

**Comment by shubhambhokare1** [comment]
> @shubhambhokare1 Are the code coverage warnings out of date? Have you already addressed them?

Yes, they are. I have addressed them previously

**Review by houseroad — APPROVED** [review]

**Comment by shubhambhokare1** [comment]
@SplitInfinity anything else required from my side for this to be merged?

**Comment by SplitInfinity** [comment]
No, I'm waiting for internal tests to pass. We had some internal tests that were failing on trunk + with this diff applied so I want to be sure that this diff does not cause additional problems.
